File: src/main/java/example/domain/model/specification/BasicFare.java
package example.domain.model.specification;

import java.util.Objects;

import example.domain.model.bill.Amount;

/**
 * 基本運賃
 */
public class BasicFare {

    Destination destination;
    public final Amount fare;

    public BasicFare(Destination destination, Amount fare) {
        this.destination = destination;
        this.fare = fare;
    }

    public Amount calculate() {
        return this.fare;
    }

    @Override
    public String toString() {
        return this.destination + "行き 料金" + fare.toString() + "";
    }

    @Override
    public boolean equals(Object other) {
        return isEqual((BasicFare) other);
    }

    private boolean isEqual(BasicFare otherFare) {
        return this.destination == otherFare.destination && this.fare.equals(otherFare.fare);
    }

    @Override
    public int hashCode() {
        return Objects.hash(destination);
    }

}

File: jr-pricing/Model/BasicFare.vb
Imports Jr.Pricing.Fw
Imports Jr.Pricing.Model.Bill
Imports Jr.Pricing.Model.Specification

Namespace Model
    ''' <summary>
    ''' 基本運賃
    ''' </summary>
    ''' <remarks></remarks>
    Public Class BasicFare : Inherits ValueObject

        Private ReadOnly destination As Destination
        Private ReadOnly fare As Amount

        Public Sub New(destination As Destination, fare As Amount)
            Me.destination = destination
            Me.fare = fare
        End Sub

        Protected Overrides Function GetAtomicValues() As IEnumerable(Of Object)
            Return New Object() {destination, fare}
        End Function

        Public Function Calculate() As Amount
            Return fare
        End Function

    End Class
End Namespace